editor/vscode/src/browser.ts

```diff
@@ -160,8 +183,16 @@ function getPreviewHtml(): string {
     window.addEventListener('message', async event => {
         if (event.data.command === "preview") {
             await render(event.data.content, event.data.base_url);
+        } else if (event.data.command === "file_loaded") {
+            let resolve = promises[event.data.url];
+            if (resolve) {
+                promises[event.data.url] = undefined;
```

Comment:
I suggest to use the delete operator to make sure that promises doesn’t have to keep the property slot around (it will grow by urls).